chain/jsonrpc/src/lib.rs

```diff
@@ -288,15 +288,17 @@ struct JsonRpcHandler {
 }
 
 impl JsonRpcHandler {
-    pub async fn process(&self, message: Message) -> Result<Message, HttpError> {
+    async fn process(&self, message: Message) -> Message {
         let id = message.id();
         match message {
-            Message::Request(request) => {
-                Ok(Message::response(id, self.process_request(request).await))
-            }
-            _ => Ok(Message::error(RpcError::parse_error(
+            Message::Request(request) => self
```

Comment:
nit: These are equivalent

```suggestion
            Message::Request(request) => Message::response(id, self.process_request(request).await)
```

Reply:
Thanks! Sorry, I've missed it